src/controllers/engine/controllerengine.cpp

```diff
@@ -223,6 +227,55 @@ void ControllerEngine::uninitializeScriptEngine() {
     }
 }
 
+void ControllerEngine::loadModule(QFileInfo moduleFileInfo) {
+    m_moduleFileInfo = moduleFileInfo;
+
+    QJSValue mod = m_pScriptEngine->importModule(moduleFileInfo.absoluteFilePath());
+    if (mod.isError()) {
+        showScriptExceptionDialog(mod);
+        return;
+    }
+
+    connect(&m_scriptWatcher,
+            &QFileSystemWatcher::fileChanged,
+            this,
+            &ControllerEngine::scriptHasChanged);
+    m_scriptWatcher.addPath(moduleFileInfo.absoluteFilePath());
+
+    QJSValue initFunction = mod.property("init");
+    if (initFunction.isCallable()) {
+        initFunction.call();
+    } else {
+        qDebug() << "Module exports no init function.";
+    }
+
+    QJSValue receiveDataFunction = mod.property("receiveData");
+    if (receiveDataFunction.isCallable()) {
+        m_receiveDataFunction = receiveDataFunction;
+    } else {
+        qWarning() << "Module exports no receiveData function.";
```

Comment:
I think this should be considered a fatal error, abort script executio and open a popup window that informs the user.